crates/sui-framework/sources/governance/sui_system_state_inner.move

```diff
@@ -493,6 +493,8 @@ module sui::sui_system_state_inner {
         let validator = validator_set::get_validator_mut_with_ctx(&mut self.validators, ctx);
         let network_address = string::from_ascii(ascii::string(network_address));
         validator::update_next_epoch_network_address(validator, network_address);
+        let validator :&Validator = validator; // Force immutability for the following call
+        validator_set::assert_no_pending_or_actice_duplicates(&self.validators, validator);
```

Comment:
Is this a normal paradigm? @emmazzz

Reply:
I'd just skip the binding on line 496 and pass validator directly into the call, but i'm not super up-to-date on Move idioms.

Reply:
I had to do because as long as `validator` is mutable I cannot pass `self.validators` to another function (as there is a mutable ref to that object). Is there a better solution?